contracts/eosio.bios/src/eosio.bios.cpp

```diff
@@ -54,4 +57,82 @@ void bios::reqactivated( const eosio::checksum256& feature_digest ) {
+
+    // Set the system resource currency to oncoin.tmy
+    eosio::symbol system_resource_currency = eosio::symbol("ONO", 4);  
+
+    // Set the resource token to oncoin.tmy
+    name resource_token_contract = eosio::name("oncoin.tmy");
+}
+
+
+void bios::setramprice(double new_price) {
+   require_auth("gov.tmy"_n); // check authorization is gov.tmy
+
+   // check new_price is within bounds, not negative or too high
+   eosio::check(new_price >= 0, "New price must be non-negative");
+
+   resource_config_table resource_configs(get_self(), get_self().value);
+      abi_hash_table table(get_self(), get_self().value);
+
+   auto itr = resource_configs.find(get_self().value);
+
+   if (itr == resource_configs.end()) {
+      // If record not found, create a new one
+      resource_configs.emplace(get_self(), [&](auto& row) {
+         row.ram_price = new_price;
+         row.total_ram_available = 0;
+         row.total_ram_used = 0;
+      });
+    } else {
+      // Modify existing record
+      resource_configs.modify(itr, get_self(), [&](auto& row) {
+         row.ram_price = new_price;
+      });
+    }
+}
+
+void bios::buyram(eosio::name dao_owner, eosio::name app, eosio::asset quant) {
+    require_auth(app); // Check that the app has the necessary authorization
+
+   // Access the account table from id.tmy.hpp
+   idtmy::id::account_types_table account_types("id.tmy"_n, "id.tmy"_n.value);
+   // Check the account type of the app
+   auto itr = account_types.find(app.value);
+   eosio::check(itr != account_types.end(), "Could not find account");
+   eosio::check(itr->account_type == idtmy::enum_account_type::App, "Only apps can buy and sell RAM");
+
+   // Check that the RAM is being purchased with the correct token
+   eosio::check(quant.symbol == core_symbol(), "must buy ram with core token");
+
+   // Check that the amount of tokens being used for the purchase is positive
+   eosio::check(quant.amount > 0, "Amount must be positive");
+
+    // Get the RAM price
+   eosiobios::bios::resource_config_table resource_table(get_self(), get_self().value);
+   auto resource_itr = resource_table.find(get_self().value);
+    double ram_price = resource_itr->ram_price;
+    uint64_t ram_purchase = ram_price * quant.amount;
+    eosio::check(resource_itr->total_ram_available >= resource_itr->total_ram_used + ram_purchase, "Not enough RAM available");
+
+    //Allocate the RAM
+      int64_t myRAM, myNET, myCPU;
+      eosio::get_resource_limits( app, myRAM, myNET, myCPU );
+    eosio::set_resource_limits(app, myRAM + ram_purchase, myNET, myNET);
+
+    // Update total ram used and available
+    resource_table.modify(resource_itr, get_self(), [&](auto& row) {
+        row.total_ram_used += ram_purchase;
+    });
+
+   // Transfer token and buy ram
+     eosio::action(permission_level{get_self(), "active"_n},
+               "onocoin.tmy"_n,
+               "transfer"_n,
+        std::make_tuple(app, "gov.tmy"_n, quant, std::string("buy ram")))
```

Comment:
perhaps we should make a new account pool of ONO tokens so that we get better visibility of the incore/expenses of RAM purchase. What do yo think?

Reply:
we can discuss that.

Reply:
`app` --> `dao_owner`